### PR TITLE
Issue #SB-17237 feat: added export telemetry

### PR DESCRIPTION
**Pull request by swetha-aggidevara**

### CODE REVIEW

File: src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.html [hunk 1]
```diff
@@ -1,70 +1,65 @@
...
-            <dt class="multiline-list-key">Last shared:</dt>
-            <dd class="multiline-list-value">19 February 2019 at 6:40 PM</dd>
+          <button type="button" class="sb-btn sb-btn-secondary sb-btn-normal
+            my-8" (click)="exportTelemetry()"
+            [disabled]="disableExport" *ngIf="telemetryInfo?.totalSize">
+            {{resourceService?.frmelmnts?.btn?.desktop?.shareTelemetry}} {{telemetryInfo?.totalSize | filesize}} </button>
+          <dl class="my-0" *ngIf="telemetryInfo?.lastExportedOn">
+            <dt class="multiline-list-key">{{resourceService?.frmelmnts?.lbl?.desktop?.lastShared}}</dt>
+            <dd class="multiline-list-value">{{telemetryInfo?.lastExportedOn |
+              date: 'd MMMM y h:mm a'}}</dd>
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.html]: at is missing from invisions

File: src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts [hunk 2]
```diff
@@ -1,15 +1,96 @@
...
 })
 export class TelemetryComponent implements OnInit {
-
-  constructor() { }
+  telemetryInfo: object = {};
+  public telemetryImpression: IImpressionEventInput;
+  public unsubscribe$ = new Subject<void>();
+  disableExport = true;
+  constructor(
+    public telemetryActionService: TelemetryActionsService,
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts]: this should be private

File: src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts [hunk 3]
```diff
@@ -1,15 +1,96 @@
...
+    public resourceService: ResourceService,
+    public toasterService: ToasterService,
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts]: this should be private

File: src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts [hunk 4]
```diff
@@ -1,15 +1,96 @@
...
+    this.getTelemetryInfo();
+    this.setTelemetryImpression();
+  }
+
+  getTelemetryInfo() {
+    this.telemetryActionService.getTelemetryInfo().subscribe(data => {
+      this.telemetryInfo = _.get(data, 'result.response');
+      this.disableExport = !this.telemetryInfo['totalSize'];
+    }, (err) => {
+      this.disableExport = true;
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts]: the error message should be shown on error

File: src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts [hunk 5]
```diff
@@ -1,15 +1,96 @@
+import { takeUntil, filter } from 'rxjs/operators';
+import { ActivatedRoute, Router, NavigationStart } from '@angular/router';
+import { TelemetryService, IImpressionEventInput } from '@sunbird/telemetry';
+import { ResourceService, ToasterService } from '@sunbird/shared';
+import { TelemetryActionsService } from './../../../offline/services';
 import { Component, OnInit } from '@angular/core';
-
+import * as _ from 'lodash-es';
```
[the remark this answers is not in the document]
Comment (swetha-aggidevara) [reply src/app/client/projects/desktop/src/app/modules/user-profile/components/telemetry/telemetry.component.ts]: If i import only get it is returning an observable

**Pull request merged**